File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/entity/Booking.ts
// Booking.ts
import {Carpooling} from "./Carpooling";
import {FeedBack} from "./FeedBack";
import {UserModel} from "./User.model";

export class Booking {
  bookingID!: number;
  nb!: number;
  carpooling!: Carpooling; // Assuming you have a Carpooling class

  user: UserModel; // Assuming you have a User class
  feedBack!: FeedBack;
// Assuming you have a FeedBack class

}

File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/entity/Booking.java
package tn.esprit.coexist.entity;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Data

public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer BookingID;

    private Integer nb;
    @ManyToOne
    @JsonIgnore
    private Carpooling carpooling;
    @ManyToOne (cascade = CascadeType.ALL)
    private User user;
    @OneToOne(mappedBy = "booking")
    @JsonIgnore
    private FeedBack feedBack;@Override
    public String toString() {
        return "Booking{" +
                "bookingID=" + getBookingID() +
                ", nb=" + nb +
                ", user=" + user + // Include other fields as needed
                ", feedback=" + feedBack + // Include other fields as needed
                '}';
    }

}
